### PR TITLE
Add a doc page for first contributors

### PR DESCRIPTION
**Pull request by charlesbvll**
#### What does this implement/fix? Explain your changes.

Add a guide for new contributors, not accustomed to the GitHub workflow.

### CODE REVIEW

File: doc/source/first-time-contributors.rst [hunk 1]
```diff
@@ -0,0 +1,314 @@
+First time contributors
+=======================
+
+This guide is geared towards people that want to get involved in Flower
+but that are not used to contributing on GitHub projects.
+
+If GitHub contributions have no secret for you,
```
Comment (danieljanes) [inline doc/source/first-time-contributors.rst:7]:
```suggestion
If you're familiar with how contributing on GitHub works, 
```

File: doc/source/first-time-contributors.rst [hunk 2]
```diff
@@ -0,0 +1,314 @@
...
+If GitHub contributions have no secret for you, 
+you can directly checkout our 
+`getting started guide for contributors <https://flower.dev/docs/getting-started-for-contributors.html>`_ 
+and examples of `good first contributions <https://flower.dev/docs/good-first-contributions.html>`_.
+
+Setting up the repository
+-------------------------
+
+1. **Create a GitHub account and setup Git**
+    Git is a distributed version control tool. This allows for an entire codebase's history to be stored and every developper's machine.
```
Comment (danieljanes) [inline doc/source/first-time-contributors.rst:16]:
```suggestion
    Git is a distributed version control tool. This allows for an entire codebase's history to be stored and every developer's machine.
```

File: doc/source/first-time-contributors.rst [hunk 3]
```diff
@@ -0,0 +1,314 @@
...
+
+    GitHub, itself, is a code hosting platform for version control and collaboration. It allows for everyone to collaborate and work from anywhere on remote repositories.
+
+    If you haven't already, you will need to create an account on `GitHub <https://github.com/signup>`_. 
+
+    The idea behind the generic Git and GitHub workflow boils down to this: 
+    you download code from a remote repository on GitHub, make changes locally and keep track of them using Git and then you upload your new history back to GitHub.
+
+2. **Forking the Flower repository**
+    A fork is a personnal copy of a GitHub repository. To create one for Flower, you must navigate to https://github.com/adap/flower (while connected to your GitHub account)
```
Comment (danieljanes) [inline doc/source/first-time-contributors.rst:27]:
```suggestion
    A fork is a personal copy of a GitHub repository. To create one for Flower, you must navigate to https://github.com/adap/flower (while connected to your GitHub account)
```

Comment (danieljanes) [reply doc/source/first-time-contributors.rst:27]: Does the URL get rendered in a clickable way?

File: doc/source/first-time-contributors.rst [hunk 4]
```diff
@@ -0,0 +1,314 @@
...
+    and click the `Fork` button situated on the top right of the page.
+
+    .. image:: _static/fork_button.png
+    
+    You can change the name if you want, but this is not necessary as this version of Flower will be yours and will sit inside your own repositories.
```
Comment (danieljanes) [inline doc/source/first-time-contributors.rst:32]:
```suggestion
    You can change the name if you want, but this is not necessary as this version of Flower will be yours and will sit inside your own account (i.e., in your own list of repositories).
```

File: doc/source/first-time-contributors.rst [hunk 5]
```diff
@@ -0,0 +1,314 @@
...
+
+4. **Add origin**
+    You can then go into the repository folder:
+
+    .. code-block:: shell
+
+        $ cd flower
+
+    And here we will need to add an origin to our repository. The origin is the \<URL\> of the remote fork repository.
+    To obtain it, we can do as previously mentionned by going to our fork repository on our GitHub account and copying the link.
```
Comment (danieljanes) [inline doc/source/first-time-contributors.rst:60]:
```suggestion
    To obtain it, we can do as previously mentioned by going to our fork repository on our GitHub account and copying the link.
```

File: doc/source/first-time-contributors.rst [hunk 6]
```diff
@@ -0,0 +1,314 @@
...
+
+    .. image:: _static/cloning_fork.png
+    
+    Once the \<URL\> is copied, we can type the following command in our terminal:
+
+    .. code-block:: shell
+
+        $ git remote add origin <URL>
```
Comment (danieljanes) [inline doc/source/first-time-contributors.rst:68]: Isn't this already set when you clone from GitHub?

File: doc/source/first-time-contributors.rst [hunk 7]
```diff
@@ -0,0 +1,314 @@
...
+
+        $ git remote add upstream https://github.com/adap/flower.git
+
+    The following diagram visually explains what we did in the previous steps:
+
+    .. image:: _static/github_schema.png 
+
+    The upstream is the GitHub remote address of the parent repository (in this case Flower), 
+    i.e. the one we eventually want to contribute to and therefore need an up-to-date history of. 
+    The origin is just the GitHub remote address of the forked repository we created.
```
Comment (danieljanes) [inline doc/source/first-time-contributors.rst:85]:
```suggestion
    The origin is just the GitHub remote address of the forked repository we created, i.e. the copy (fork) in our own account.
```

File: doc/source/first-time-contributors.rst [hunk 8]
```diff
@@ -0,0 +1,314 @@
...
+
+    Once you click the `Compare & pull request` button, you should see something similar to this:
+
+    .. image:: _static/creating_pr.png
+    
+    At the top you have an explanation of which branch will be merged where:
+
+    .. image:: _static/merging_branch.png
+    
+    In this example you can see that the request is to merge the branch `doc-fixes` from my forked repository to branch main from the Flower repository.
```
Comment (danieljanes) [inline doc/source/first-time-contributors.rst:191]:
```suggestion
    In this example you can see that the request is to merge the branch `doc-fixes` from my forked repository to branch ``main`` from the Flower repository.
```

**Review by danieljanes — CHANGES_REQUESTED** [review]
I added a few minor comments, but overall it looks really good!

File: doc/source/first-time-contributors.rst (hunk 3 above)
Comment (charlesbvll) [reply doc/source/first-time-contributors.rst:27]: Yes it does

File: doc/source/first-time-contributors.rst (hunk 6 above)
Comment (charlesbvll) [reply doc/source/first-time-contributors.rst:68]: Yes indeed, but it felt easier to understand the whole workflow by doing it explicitly. I can either remove the step or just mark it as optional, what do you think is best?

Comment (danieljanes) [reply doc/source/first-time-contributors.rst:68]: Makes sense, let's leave it in there

**Review by danieljanes — APPROVED** [review]
Lgtm!

**Pull request merged**